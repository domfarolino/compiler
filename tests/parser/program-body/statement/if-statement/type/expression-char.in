program dominic is

  char expression;

begin

  expression := 'c';

  if (expression) then
    expression := 'd';
  end if;

end program.
