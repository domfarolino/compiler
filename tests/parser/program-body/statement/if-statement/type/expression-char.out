[1;31mLine 9: If statement must be a boolean or boolean equivalent, not char
Line 9: Error parsing program body
[0m